base_local_planner/include/base_local_planner/obstacle_cost_function.h

```diff
@@ -61,9 +61,10 @@ class ObstacleCostFunction : public TrajectoryCostFunction {
 
   void setSumScores(bool score_sums){ sum_scores_=score_sums; }
 
-  void setParams(double max_trans_vel, double max_forward_inflation, double max_sideward_inflation, double scaling_speed, bool occdist_use_footprint = true);
+  void setParams(double max_trans_vel, double max_forward_inflation, double max_sideward_inflation,
+                 double scaling_speed, bool occdist_use_footprint = true, double scaling_discount_factor = 1.0);
```

Comment:
Q/S: can we invert the order here? I think it makes more sense and the `doubles` are together. I would remove defaults too
```suggestion
                 double scaling_speed, double scaling_discount_factor = 1.0, bool occdist_use_footprint = true);
```

Reply:
I switched the order.

This function exists in original DWA so I added the defaults to keep backwards compatibility